analysis_options.yaml

```diff
@@ -1,2 +1,215 @@
+    - no_adjacent_strings_in_list
+    - no_duplicate_case_values
+    - no_logic_in_create_state
+    - prefer_relative_imports      # personal preference
+    - prefer_void_to_null
+    - test_types_in_equals
+    - throw_in_finally
+    - unnecessary_statements
+    - unrelated_type_equality_checks
+    - unsafe_html
+    - use_key_in_widget_constructors
+    - valid_regexps
+
+## Style Rules
+
+    - always_declare_return_types
+#    - always_put_control_body_on_new_line # works against dartfmt
+    - always_put_required_named_parameters_first
+    - always_require_non_null_named_parameters
+#    - always_specify_types        # this works agains general Dart style guides
+    - annotate_overrides
+    - avoid_annotating_with_dynamic
+#    - avoid_as                    # obsolete - this was considered good practice in Dart 1
+    - avoid_bool_literals_in_conditional_expressions
+    - avoid_catches_without_on_clauses
+    - avoid_catching_errors
+# Annoying rule when using classes as namespaces for constants which is discouraged by the Dart
+# style guide, but I prefer it over aliases in imports for disambiguation.
+#    - avoid_classes_with_only_static_members
+    - avoid_double_and_int_checks
+    - avoid_equals_and_hash_code_on_mutable_classes
+    - avoid_escaping_inner_quotes  # TODO(zoechi) not yet available
+    - avoid_field_initializers_in_const_classes
+    - avoid_function_literals_in_foreach_calls
+    - avoid_implementing_value_types
+    - avoid_init_to_null
+    - avoid_js_rounded_ints
+    - avoid_null_checks_in_equality_operators
+    - avoid_positional_boolean_parameters
+    - avoid_private_typedef_functions
+    - avoid_redundant_argument_values
+    - avoid_renaming_method_parameters
+    - avoid_return_types_on_setters
+    - avoid_returning_null
+    - avoid_returning_null_for_void
+    - avoid_returning_this
+    - avoid_setters_without_getters
+    - avoid_shadowing_type_parameters # personal preference
+    - avoid_single_cascade_in_expression_statements
+    - avoid_types_on_closure_parameters
+    - avoid_unnecessary_containers
+    - avoid_unused_constructor_parameters
+    - avoid_void_async
+    - await_only_futures
+    - camel_case_extensions
+    - camel_case_types
+    - cascade_invocations
+    - constant_identifier_names
+    - curly_braces_in_flow_control_structures # personal preference
+    - directives_ordering
+    - do_not_use_environment # depends on project requirements
+    - empty_catches
+    - empty_constructor_bodies
+    - exhaustive_cases
+    - file_names
+#    - flutter_style_todos         # I find the required ":" annoying and pointless
+    - implementation_imports
+    - join_return_with_assignment
+    - leading_newlines_in_multiline_strings
+    - library_names
+    - library_prefixes
+    - lines_longer_than_80_chars
+# Too annoying when long strings with interpolations are broken into multiple lines which we do often
+#    - missing_whitespace_between_adjacent_strings
```

Comment:
@zoechi isn't it helpful to avoid missing spaces?

Reply:
I'll check again.
I remember this warning about code that's actually fine.

Reply:
I'm enabling this for now as this wouldn't affect current source in anyway, can be disabled later if it doesn't fit

c59284f2906efdd2cf1c6bc7a14125db33d5b37f